centinel/primitives/tls.py

```diff
@@ -22,6 +23,11 @@ def get_fingerprint(host, port=443, external=None):
         tls_error = str(exp)
         exception = exp
 
+    # this comes out as unicode, but m2crypto breaks if it gets
+    # something other than a string, so convert to ascii
+    if type(cert) == unicode or test:
```

Comment:
@ben-jones `or test:`? What does this part do? It will throw an exception if `test` is not declared.